### PR TITLE
fix various timing issues

### LINKED ISSUE #90: Issues with validations both in frontend and backend
1.Complaints at odd times eg 2 am to 6 am being accepted.
2.Complaints with Sundays and saturdays as availbility days being accepted.
3.Complaints with avail_time on the same day before filing the complaint being accepted.

### PR DESCRIPTION
**Pull request by Kushagra-0801**
fix #90

### CODE REVIEW

File: complaints/models.py [hunk 1]
```diff
@@ -116,25 +118,30 @@ def validate_timings(self, avail_start_time, avail_end_time):
...
                 if avail_start_time >= start_time and avail_end_time <= end_time:
-                    break
+                    return
             else:
                 raise ValidationError(
                     "Girls' Hostel is only open for limited time slots. Your given time does not fit in them"
                 )
+        for start_time, end_time in self.TECHNICIAN_WORK_HOURS:
+            if avail_start_time >= start_time and avail_end_time <= end_time:
+                break
```
Comment (Aviral14) [inline complaints/models.py]: Don't be too strict with the timings.Even if the availibility timings are from 9 am to 2 pm they should be accepted.But if the user enters the timings which completely lie in the technicians lunch hour ,then only an error message should b sent.

File: complaints/models.py [hunk 2]
```diff
@@ -116,25 +118,30 @@ def validate_timings(self, avail_start_time, avail_end_time):
...
-            ):
-                raise ValidationError("Available time is less than one hour.")
+            raise ValidationError("Availble time not in technicians' working hours")
+        if self.avail_date == date.today():
+            if avail_start_time < datetime.now().time():
+                raise ValidationError("Available time before complaint registration.")
+        if avail_end_time < time(
+            avail_start_time.hour + 1, avail_start_time.minute, avail_start_time.second
+        ):
+            raise ValidationError("Available time is less than one hour.")
```
Comment (Aviral14) [inline complaints/models.py]: In every validation error add the message  " Complaint not registered .....  Try again! "

File: complaints/models.py (hunk 1 above)
Comment (Kushagra-0801) [reply complaints/models.py]: I don't know fuzzy logic dude. Give me hard values

Comment (Aviral14) [reply complaints/models.py]: This isn't any fuzzy logic dude!!! My comment is quite easy to understand.Understand it to contemplate the hard rules yourself!

Comment (Kushagra-0801) [reply complaints/models.py]: I don't get it. I need some help senpai

**Pull request merged**